File: Arrays/Remove Element/cpp/swap_with_last_solution.cpp
#include <iostream>
#include <vector>
using namespace std;

int removeElement(vector<int>& nums, int val) {
    int i = 0;
    int n = nums.size();
    while (i < n) {
        if (nums[i] == val) {
            nums[i] = nums[n-1]; // swap with last element
            n--; // shrink array
        } else {
            i++;
        }
    }
    return n; // number of valid elements
}

File: Arrays/Remove Element/python /swap_with_last_solution.py
def remove_element(nums, val):
    i = 0
    n = len(nums)
    while i < n:
        if nums[i] == val:
            # Swap with last element
            nums[i] = nums[n-1]
            n -= 1  # reduce size of valid array
        else:
            i += 1  # move to next element
    return n  # number of valid elements
